TypeScript Package

PaulRBerg opened the pull request:
See also the associated PR in the docs repo: https://github.com/sablier-labs/docs/pull/308

- Closes sablier-labs/deployments#21
- Closes sablier-labs/deployments#24
- Closes sablier-labs/deployments#25
- Closes sablier-labs/deployments#27
- Closes sablier-labs/sdk#8
- Closes sablier-labs/sdk#7

PaulRBerg committed:
feat: typescript package with deployment data

Massive refactor

See https://github.com/sablier-labs/deployments/issues/21

PaulRBerg commented:
The PR is finally ready for review, @sablier-labs/evm!

We should hop on a call to discuss next week. Feel free to book a time in my Calendly.

PaulRBerg committed:
refactor: queries contracts get

smol-ninja commented:
Well done. I have booked for Tuesday afternoon.

PaulRBerg committed:
feat: add "default" field in Chain type

PaulRBerg committed:
build: bump biome dep

PaulRBerg committed:
refactor: shorter name for resolvers

chore: more comments
chore: remove unneeded fields in tsconfig

PaulRBerg committed:
ai: configure repomix

PaulRBerg committed:
chore: repomix

PaulRBerg committed:
test: deployment blocks for indexed contracts

PaulRBerg committed:
test: reorganize the tests

PaulRBerg committed:
test: fix broken tests for zk broadcasts

feat: more aliases in justfile
feat: script for printing chains
fix: fix addresses for Sophon
refactor: rename scripts
test: rename functions

PaulRBerg committed:
refactor: change export structure of releases

docs: draft README and CONTRIBUTING guide
refactor: rename queries to sablier
refactor: fewer default exports

PaulRBerg committed:
fix: getter in release queries

build: add "viem" as dependency

PaulRBerg committed:
refactor: rename data broadcast dirs

PaulRBerg committed:
refactor: delete indexers

build: move build-only settings to "tsconfig.build.json"
refactor: export mainnets before testnets
refactor: get rid of "logAndThrow"
refactor: move release-specific helpers
refactor: move "logger" and "check-broadcast" to "src/internal"
refactor: non-default exports
refactor: rename "scripts" to "cli"
refactor: simplify logic in "resolvers.ts"

PaulRBerg committed:
build: bump version

PaulRBerg committed:
build: bump version

smol-ninja commented:
Since, its very difficult to post comments on the diff, I will post some of them here (which requires discussion) and the rest can be seen in my PR: https://github.com/sablier-labs/deployments/pull/41.

Its not a good idea to emit the entire periphery deployment under airdrops repo. Between `SABLIER_V2_BATCH` and `SABLIER_V2_MERKLE_STREAMER_FACTORY`, only the latter should be a part of the airdrops release whereas `SABLIER_V2_BATCH` should be part of the lockup release. We can fix it by refactoring as the following:

```ts
    core: {
      [manifest.core.SABLIER_V2_COMPTROLLER]: "0x17Ec73692F0aDf7E7C554822FBEAACB4BE781762",
      [manifest.core.SABLIER_V2_LOCKUP_DYNAMIC]: ["0xf390cE6f54e4dc7C5A5f7f8689062b7591F7111d", 161_612_601],
      [manifest.core.SABLIER_V2_LOCKUP_LINEAR]: ["0xFDD9d122B451F549f48c4942c6fa6646D849e8C1", 161_611_816],
      [manifest.core.SABLIER_V2_NFT_DESCRIPTOR]: "0x2fb103fC853b2F5022a840091ab1cDf5172E7cfa",
      [manifest.periphery.SABLIER_V2_BATCH]: "0xAFd1434296e29a0711E24014656158055F00784c",
    },
    periphery: {
      [manifest.periphery.SABLIER_V2_MERKLE_STREAMER_FACTORY]: [
        "0x237400eF5a41886a75B0e036228221Df075b3B80",
        161_026_555,
      ],
```

If you agree with this, lmk and I will be happy to make this change in my review PR.

PaulRBerg committed:
build: bump viem

build: move decorator settings in tsconfig
chore: ignore node_modules in biome config
chore: ignore empty message logs in winston
refactor: refactor "logInfo" and "logVerbose" into "log"

PaulRBerg committed:
feat: more rpc infura and alchemy

PaulRBerg committed:
build: beta release

fix: fix circular dependency

PaulRBerg committed:
feat: allow chain id in contracts queries

smol-ninja committed:
Review of PR #31 (#41)

* feat: recursive approach instead of hardcoding protocols
docs: natspec of deployment query
fix: reverts when only chainId is passed in getAll function
chore: adds missing rpc endpoints
fix: tangle native token
feat: adds missing parameters in CompilerSettings type
fix: invalid undefined return when flatMap is used
refactor: alphabetical order of Flow deployments
fix: incorrect isLatest value for older Airdrops deployments
fix: missing zksync broadcasts
script: default just recipe

* refactor: allow chainId in getAll

* refactor: simpler definition for catalog

---------

Co-authored-by: Paul Razvan Berg <prberg@proton.me>

PaulRBerg committed:
refactor: simplify airdrops v1.1 and v1.2 exports

smol-ninja commented:
You are right. I misread that. All good now. Thanks.

smol-ninja committed:
docs: update natspecs of contracts getAll query

PaulRBerg committed:
docs: roll beta.2

PaulRBerg committed:
refactor: proper cli with commander

fix: fix path in "checkBroadcast"

PaulRBerg committed:
ci: fix devkit

PaulRBerg committed:
refactor: rename package to sablier

PaulRBerg committed:
chore: update description

PaulRBerg commented:
I was almost ready to merge this PR, but I realized that what I built is so different from what this repo was originally meant to be that I decided to put my work into a new repo called SDK:

https://github.com/sablier-labs/sdk

The package will be available on npm as [`sablier`](https://npmjs.com/package/sablier) 🤩

---

Besides that, there is another benefit to creating a new repo (SDK) and archiving this repo (deployments):

Backward-compatibility with the links we have shared over time, e.g. in the docs, we use the `main` branch as a re  instead of a particular git commit ref. See an example [here](https://github.com/sablier-labs/docs/blob/9f2fba0fb3c230df69424490019319dbebb68b91/docs/guides/flow/previous-deployments/v1.0.md?plain=1#L30)).
